### PR TITLE
Source google-analytics-data-api: upgrade cdk 3

### PR DESCRIPTION
**Pull request by aldogonzalez8**
## What
The google-analytics-data-api currently uses a few methods should_retry() and backoff_time() which are slated to be deprecated in https://github.com/airbytehq/airbyte/pull/39811

This is related to green weeks because this connector currently raises many system errors for 429 rate limited, which should be recategorized as transient errors. This behavior should be accessible when the legacy Python CDK HttpStream class starts using the new HttpClient. But to do so, we need to account for updating the connector in response to the breaking change from 2.0.0 to 3.0.0.

Resolves https://github.com/airbytehq/airbyte-internal-issues/issues/8576
## How
Bump the CDK version to the latest, deprecate old methods, implement an error handler, and try to keep the current error messaging functionality.

## Review guide

1. `airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py`: 
   a.- Backoff strategy and error handler are implemented in **GoogleAnalyticsDataApiAbstractStream**.
   b.- _should_retry_ and _backoff_time_ are removed from **GoogleAnalyticsDataApiAbstractStream**
   c.- _get_error_mapping_ is implemented in **GoogleAnalyticsDataApiAbstractStream** to handle independent mappers on child streams.
   d.- _get_error_mapping_ is implemented in **GoogleAnalyticsDataApiBaseStream**. Previously, _check_connection_ in **SourceGoogleAnalyticsDataApi** would format a config error (WRONG_CUSTOM_REPORT_CONFIG) for HTTP errors checking the connection (read records). Now method _get_google_analytics_data_api_base_error_mapping_ handles this task.
   e.- _get_error_mapping_ is implemented in **GoogleAnalyticsDataApiMetadataStream**. Previously, _check_connection_ in **SourceGoogleAnalyticsDataApi** would format and raise an "Incorrect Property ID:..." message for 400, 403 HTTP exceptions checking the connection (read records). Now the method _get_google_analytics_data_api_metadata_error_mapping_ handles this task.
2. `airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/api_quota.py`: remove should_retry and raise_on_http_errors properties and add response_action we can use in GoogleAnalyticsDatApiErrorHandler.interpret_response. 
3. For test files I had to make changes to remove the deprecated attribute checks, include the new ones and some required changes for error formats. 

## User Impact
Change should be as transparent as possible for final users, but 429 should now be transient errors.

## Can this PR be safely reverted and rolled back?
* If unsure, leave it blank.
- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/google_analytics_data_api_base_error_mapping.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        if base_error_resolution.failure_type in (FailureType.config_error, FailureType.system_error):
+            error_resolution = ErrorResolution(
+                response_action=base_error_resolution.response_action,
+                failure_type=FailureType.config_error,
+                error_message=WRONG_CUSTOM_REPORT_CONFIG.format(report=report_name),
+            )
+            stream_error_mapping[error_key] = error_resolution
+        else:
+            stream_error_mapping[error_key] = base_error_resolution
+    return stream_error_mapping
```
Comment (aldogonzalez8) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/google_analytics_data_api_base_error_mapping.py:24]: d. 
get_error_mapping is implemented in **GoogleAnalyticsDataApiBaseStream**. 

Previously, check_connection in **SourceGoogleAnalyticsDataApi** would format a config error (WRONG_CUSTOM_REPORT_CONFIG) for HTTP errors checking the connection (read records). Now method _get_google_analytics_data_api_base_error_mapping_ handles this task.

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py [hunk 2]
```diff
@@ -138,6 +159,9 @@ class GoogleAnalyticsDataApiBaseStream(GoogleAnalyticsDataApiAbstractStream):
 
     metadata = MetadataDescriptor()
 
+    def get_error_mapping(self) -> Mapping[Union[int, str, Type[Exception]], ErrorResolution]:
+        return get_google_analytics_data_api_base_error_mapping(self.name)
```
Comment (aldogonzalez8) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py:163]: d.- get_error_mapping is implemented in **GoogleAnalyticsDataApiBaseStream**. 

Previously, check_connection in **SourceGoogleAnalyticsDataApi** would format a config error (WRONG_CUSTOM_REPORT_CONFIG) for HTTP errors checking the connection (read records). Now method _get_google_analytics_data_api_base_error_mapping_ handles this task.

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py [hunk 3]
```diff
@@ -561,11 +580,8 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
                 try:
                     stream_slice = next(report_stream.stream_slices(sync_mode=SyncMode.full_refresh))
                     next(report_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice), None)
-                except HTTPError as e:
-                    error_response = ""
-                    if e.response.status_code == HTTPStatus.BAD_REQUEST:
-                        error_response = e.response.json().get("error", {}).get("message", "")
-                    return False, WRONG_CUSTOM_REPORT_CONFIG.format(report=report["name"], error_response=error_response)
+                except MessageRepresentationAirbyteTracedErrors as e:
+                    return False, f"{e.message}. {self._extract_message_details(e.internal_message)}"
```
Comment (aldogonzalez8) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py:584]: d.- get_error_mapping is implemented in **GoogleAnalyticsDataApiBaseStream**. 

Previously, check_connection in **SourceGoogleAnalyticsDataApi** would format a config error (WRONG_CUSTOM_REPORT_CONFIG) for HTTP errors checking the connection (read records). Now method _get_google_analytics_data_api_base_error_mapping_ handles this task.

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/google_analytics_data_api_metadata_error_mapping.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            response_action=ResponseAction.FAIL,
+            failure_type=FailureType.config_error,
+            error_message=MESSAGE.format(property_id=property_id, property_id_docs_url=PROPERTY_ID_DOCS_URL),
+        ),
+        400: ErrorResolution(
+            response_action=ResponseAction.FAIL,
+            failure_type=FailureType.config_error,
+            error_message=MESSAGE.format(property_id=property_id, property_id_docs_url=PROPERTY_ID_DOCS_URL),
+        ),
+    }
```
Comment (aldogonzalez8) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/google_analytics_data_api_metadata_error_mapping.py:26]: e.- 
_get_error_mapping_ is implemented in **GoogleAnalyticsDataApiMetadataStream**. 

Previously, _check_connection_ in **SourceGoogleAnalyticsDataApi** would format and raise an "Incorrect Property ID:..." message for 400, 403 HTTP exceptions checking the connection (read records). Now the method _get_google_analytics_data_api_metadata_error_mapping_ handles this task.

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py [hunk 5]
```diff
@@ -431,6 +455,9 @@ def path(
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
         yield response.json()
 
+    def get_error_mapping(self):
+        return get_google_analytics_data_api_metadata_error_mapping(self.config.get("property_id"))
+
```
Comment (aldogonzalez8) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py:460]: e.- 
_get_error_mapping_ is implemented in **GoogleAnalyticsDataApiMetadataStream**. 

Previously, _check_connection_ in **SourceGoogleAnalyticsDataApi** would format and raise an "Incorrect Property ID:..." message for 400, 403 HTTP exceptions checking the connection (read records). Now the method _get_google_analytics_data_api_metadata_error_mapping_ handles this task.

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py [hunk 6]
```diff
@@ -523,19 +550,11 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
...
-
-                    wrong_property_id_error = AirbyteTracedException(
-                        message=message, internal_message=internal_message, failure_type=FailureType.config_error
-                    )
-                    raise wrong_property_id_error
+            except (MessageRepresentationAirbyteTracedErrors, BaseBackoffException) as ex:
+                if hasattr(ex, "failure_type") and ex.failure_type == FailureType.config_error:
+                    # bad request and forbidden are set in mapper as config errors
+                    raise ex
+                logger.error(f"Check failed", exc_info=ex)
```
Comment (aldogonzalez8) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py:557]: e.- 
_get_error_mapping_ is implemented in **GoogleAnalyticsDataApiMetadataStream**. 

Previously, _check_connection_ in **SourceGoogleAnalyticsDataApi** would format and raise an "Incorrect Property ID:..." message for 400, 403 HTTP exceptions checking the connection (read records). Now the method _get_google_analytics_data_api_metadata_error_mapping_ handles this task.

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py [hunk 7]
```diff
@@ -561,11 +580,8 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
                 try:
                     stream_slice = next(report_stream.stream_slices(sync_mode=SyncMode.full_refresh))
                     next(report_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice), None)
-                except HTTPError as e:
-                    error_response = ""
-                    if e.response.status_code == HTTPStatus.BAD_REQUEST:
-                        error_response = e.response.json().get("error", {}).get("message", "")
-                    return False, WRONG_CUSTOM_REPORT_CONFIG.format(report=report["name"], error_response=error_response)
+                except MessageRepresentationAirbyteTracedErrors as e:
+                    return False, f"{e.message} {self._extract_internal_message_error_response(e.internal_message)}"
```
Comment (aldogonzalez8) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py:584]: d.- get_error_mapping is implemented in **GoogleAnalyticsDataApiBaseStream**. 

Previously, check_connection in **SourceGoogleAnalyticsDataApi** would format a config error (WRONG_CUSTOM_REPORT_CONFIG) for HTTP errors checking the connection (read records). Now method _get_google_analytics_data_api_base_error_mapping_ handles this task.

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/utils.py [hunk 8]
```diff
@@ -71,7 +71,7 @@
 WRONG_METRICS = "The custom report {report_name} entered contains invalid metrics: {fields}. Validate your custom query with the GA 4 Query Explorer (https://ga-dev-tools.google/ga4/query-explorer/)."
 WRONG_PIVOTS = "The custom report {report_name} entered contains invalid pivots: {fields}. Ensure the pivot follow the syntax described in the docs (https://developers.google.com/analytics/devguides/reporting/data/v1/rest/v1beta/Pivot)."
 API_LIMIT_PER_HOUR = "Your API key has reached its limit for the hour. Wait until the quota refreshes in an hour to retry."
-WRONG_CUSTOM_REPORT_CONFIG = "Please check configuration for custom report {report}. {error_response}"
+WRONG_CUSTOM_REPORT_CONFIG = "Please check configuration for custom report {report}."
```
Comment (aldogonzalez8) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/utils.py:74]: d.- get_error_mapping is implemented in GoogleAnalyticsDataApiBaseStream.

Previously, check_connection in SourceGoogleAnalyticsDataApi would format a config error (WRONG_CUSTOM_REPORT_CONFIG) for HTTP errors checking the connection (read records). Now method get_google_analytics_data_api_base_error_mapping handles this task.

Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/utils.py:74]: The internal error_response is added to message when exception is cached

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py (hunk 7 above)
Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py:584]: The internal error_response is added to message when exception is cached

**Comment by aldogonzalez8** [comment]
This seems a problem for acceptance testing 
image

**Comment by aldogonzalez8** [comment]
Currently working on timing outs for a full refresh in acceptance.

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/api_quota.py [hunk 9]
```diff
@@ -105,16 +99,14 @@ def _get_known_quota_from_response(self, property_quota: Mapping[str, Any]) -> M
     def _set_retry_attrs_for_quota(self, quota_name: str) -> None:
         quota = self.quota_mapping.get(quota_name, {})
         if quota:
-            self.should_retry = quota.get("should_retry")
-            self.raise_on_http_errors = quota.get("raise_on_http_errors")
+            self.response_action = quota.get("response_action")
             self.stop_iter = quota.get("stop_iter")
             self.backoff_time = quota.get("backoff")
-            self.error_message = quota.get("error_message")
+            self.error_message = quota.get("error_message") or quota.get("error_pattern")
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/api_quota.py:105]: nit:
```python
self.error_message = quota.get("error_message",  quota.get("error_pattern"))
```

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/google_analytics_data_api_base_error_mapping.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+from airbyte_cdk.sources.streams.http.error_handlers.default_error_mapping import DEFAULT_ERROR_MAPPING
+from airbyte_cdk.sources.streams.http.error_handlers.response_models import ErrorResolution
+from source_google_analytics_data_api.utils import WRONG_CUSTOM_REPORT_CONFIG
+
+
+def get_google_analytics_data_api_base_error_mapping(report_name) -> Mapping[Union[int, str, Type[Exception]], ErrorResolution]:
+    stream_error_mapping = {}
+    for error_key, base_error_resolution in DEFAULT_ERROR_MAPPING.items():
+        if base_error_resolution.failure_type in (FailureType.config_error, FailureType.system_error):
+            error_resolution = ErrorResolution(
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/google_analytics_data_api_base_error_mapping.py:16]: nit:
```python
stream_error_mapping[error_key] = ErrorResolution(
                response_action=base_error_resolution.response_action,
                failure_type=FailureType.config_error,
                error_message=WRONG_CUSTOM_REPORT_CONFIG.format(report=report_name),
            )
```

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py [hunk 11]
```diff
@@ -85,18 +89,41 @@ def __get__(self, instance, owner):
...
+                error_message=GoogleAnalyticsQuotaHandler.error_message,
+            )
+        return super().interpret_response(response_or_exception)
+
+
 class GoogleAnalyticsDataApiAbstractStream(HttpStream, ABC):
     url_base = "https://analyticsdata.googleapis.com/v1beta/"
     http_method = "POST"
-    raise_on_http_errors = True
+    _config: Mapping[str, Any] = {}
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py:118]: Could you please leave the comment, why do we need this?

**Review by bazarnov — COMMENTED** [review]
Some complex work has been done here, good work @aldogonzalez8 ! Left few comments to take a look.

**Comment by bazarnov** [comment]
> This seems a problem for acceptance testing image

This is known and expected, you have to be patient with `Google Analytics` sources)

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py (hunk 11 above)
Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py:118]: At some stage of my development when running the UT I had an error message stating it couldn't resolve "_config" but I just removed this line and the error is not present so I can just assume it was something different that was causing that message, anyway will remove this and resolve the rest of your comments. Thanks!

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/google_analytics_data_api_base_error_mapping.py [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+    for error_key, base_error_resolution in DEFAULT_ERROR_MAPPING.items():
+        if base_error_resolution.failure_type in (FailureType.config_error, FailureType.system_error):
+            stream_error_mapping[error_key] = ErrorResolution(
+                response_action=base_error_resolution.response_action,
+                failure_type=FailureType.config_error,
+                error_message=WRONG_CUSTOM_REPORT_CONFIG.format(report=report_name),
+            )
+        else:
+            stream_error_mapping[error_key] = base_error_resolution
+    return stream_error_mapping
```
Comment (aldogonzalez8) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/google_analytics_data_api_base_error_mapping.py:26]: d.- get_error_mapping is implemented in **GoogleAnalyticsDataApiBaseStream**. 

Previously, check_connection in **SourceGoogleAnalyticsDataApi** would format a config error (WRONG_CUSTOM_REPORT_CONFIG) for HTTP errors checking the connection (read records). Now method _get_google_analytics_data_api_base_error_mapping_ handles this task.

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/google_analytics_data_api_metadata_error_mapping.py [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
```
Comment (aldogonzalez8) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/google_analytics_data_api_metadata_error_mapping.py:29]: e.- 
_get_error_mapping_ is implemented in **GoogleAnalyticsDataApiMetadataStream**. 

Previously, _check_connection_ in **SourceGoogleAnalyticsDataApi** would format and raise an "Incorrect Property ID:..." message for 400, 403 HTTP exceptions checking the connection (read records). Now the method _get_google_analytics_data_api_metadata_error_mapping_ handles this task.

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/google_analytics_data_api_base_error_mapping.py (hunk 10 above)
Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/google_analytics_data_api_base_error_mapping.py]: Fixed

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py (hunk 11 above)
Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py]: Fixed

**Comment by aldogonzalez8** [comment]
I am crossing my fingers that there are enough tokens this time to complete CAT; if I get it, I will give it a few minutes to run regression as I assume it will consume tokens from the same account.

**Comment by aldogonzalez8** [comment]
Ok, I was finally able to get regression testing with the current records count for the target, main problems were:

1. Pivot_report: For **GoogleAnalyticsDataApiBaseStream**._stream_slices_ method, we only support full_refresh for the pivot report, so there was no problem when there was no cursor available and trying to get start_date from the state. Thanks, @bazarnov, for pointing this out.
2. Cohort_report: the CohortReportMixin.stream_slices method was returning `yield from [None],` with cdk upgrade when reading slices; this won't play well as the current validation while reading is while [next_slice is not None:](https://github.com/airbytehq/airbyte/blob/aa5d6dd6b342e8bd65cc766a2cefaac5f6ab071b/airbyte-cdk/python/airbyte_cdk/sources/streams/core.py#L177) the stream results were 0. So, I had to use [this hack](https://github.com/airbytehq/airbyte/blob/aa5d6dd6b342e8bd65cc766a2cefaac5f6ab071b/airbyte-cdk/python/airbyte_cdk/sources/streams/core.py#L439-L444) that seems made feel bad to @brianjlai.

**Comment by aldogonzalez8** [comment]
Regression testing from my local because if there are no pics it didn't happen

Note: same records with fewer requests. I assume this is because the availability strategy is gone. Also, it says catalogs are different, but this is because now we have fields like is_resumable.

image
image
image
image

**Comment by aldogonzalez8** [comment]
And yes, the catalog differences when running discovery are because of the "is_resumable" field added to the streams schema.

image

**Comment by aldogonzalez8** [comment]
CAT ran out of quota, will give a couple of hours before try again.

**Comment by aldogonzalez8** [comment]
Running cat again...

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py [hunk 14]
```diff
@@ -85,18 +89,40 @@ def __get__(self, instance, owner):
...
+
+class GoogleAnalyticsDatApiErrorHandler(HttpStatusErrorHandler):
+    @GoogleAnalyticsQuotaHandler.handle_quota()
+    def interpret_response(self, response_or_exception: Optional[Union[requests.Response, Exception]] = None) -> ErrorResolution:
+        if response_or_exception.status_code == requests.codes.too_many_requests:
+            return ErrorResolution(
+                response_action=GoogleAnalyticsQuotaHandler.response_action,
+                failure_type=FailureType.transient_error,
+                error_message=GoogleAnalyticsQuotaHandler.error_message,
+            )
```
Comment (aldogonzalez8) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py:111]: This is related to green weeks because this connector currently raises a lot of system errors for 429 rate limited which should actually be recategorized as transient errors.

**Comment by aldogonzalez8** [comment]
Here is the regression testing GHA [run](https://github.com/airbytehq/airbyte/actions/runs/10198131805/job/28212401938) and [test results](https://ffbf67c902bc85978de96565b49ed0e8917d7e9a4658ae601d9443f-apidata.googleusercontent.com/download/storage/v1/b/airbyte-ci-reports-multi/o/airbyte-ci%2Fconnectors%2Ftest%2Fmanual%2Faldogonzalez8_source_google-analytics-data-api_upgrade-cdk-3%2F1722514719%2F952c34b002c8bfb9abd0ad1574c222adc3f1f5a3%2Fsource-google-analytics-data-api%2F2.4.15%2Foutput.html?jk=AYBlUPCOeICaSccDNujqZPr3clJXzzgNav4oOY3TCWH1YgOv0grlIxbV6iV4-KGmsHMW6ZCeBc-6VhnsHsaEI6VMN86RZQA_4hHmIuoOzSbAqnFcGVBWdO9ROiUWdTGUmDDQinOLN64fYvxfMWnNoWwHOlNnCSa5rXb2cYfrvQIW9wGGUR_xjU513KV83KqI41kt7axJMcY_n1DD3cajHLVmCkSqe3JCxqQRe5sbC2RKDvD-cGJuA4bT9N_Ou1VCk0mS_3Tp05CvkpHTg7otWKmh2SV6YRtzwNU3NLZ1P8Yy3bowTPZMk-D4ImyXKxYqMqikExF3mDYEhW63i3etlWwPVNUSmqNf6gIOFZmlxo1UGGhuulpA5bpF40RwNVZxQ6UpnSroHwWc76R6D4BvaOBGMtJYF9Hebtnp0RdtU-R-LgEdsmZiA927F8o2NbF-igKdBGgp4qEBzH6dxITp9-a02fWkyKVik_gYjxC43qHLhP-oxjJPdCL8HWg8LZOIg9Sbdy_w_la6Suyn6qcBi_V-HfYx7Ld6Hq2ixiziTeGk62vUrB8V8FwnTU6H0ZNR0tCCaSeyd6nNkWvPs_natxZtilhqKURy6O2IpL60vsVGaUJxwchf1pdrXKqfdQvniqpt0Kje-JSpNWsWbXCgrFMmQh89NxeEDIHNuoW913staQkIMxqxaAcWa0TOIaRpHzIkFARLPF9rgpe4gQjVEeJfImemxr3HXB0vSSyHH7WAAGh_0ZyPAERGW3mhGtktuTCcmQU2u40a7RBDfmaIT52QqQBt1zf04gIG30qIpMHMWPQPooWo7zuP2RXDVUEZ_ZH_psy1ri0bQt1C5Xvmu72XNyAs6X8i5MXSVnFr1QJQGhWuTiMGk6JduipLlSBUPKr4iDQEdglP6pjxVvjQmUdJ_JZUwtRFT8UpEKapT1y8p8_g5vx6g7CSzDfiglDwXioNUoQLKfNTeFt1iHQcO1g12w6z61LEl8Trd5lhRGb--mNrUxM5j4wqcNrQekxUj4xCoE-5ZjRO3SlWact3f2pBU9DmdUSS8qdIJJl8ytCEfKJmo8M4taJlbCY8DrBTq048DWxHAnqA0pINjbaG41__w8hNGfvu7In-AOfJBRSHJmNB9UM-rmRbqEzV6qS3ZFiciF3TEMY5tAiYV2pgCybkB8U8LGTIRcQMYuwnek6ggXI0NNwz52G3hKST5fnjRQHzEESl8RBOlo94uCJEMU1EyinDzineV9ziZb3hwFC2p9ZZSQhxHWr_aY5YBo4XLO6C8P6IBx7kVJ-EM4flFBKnXN56y8EPbQ6xoX2LRAe-VpO9HP6JKk9TrzGjPTSihIR0KvZi0z0Rsfaz4Szuap-fdzadMd_j_ePSuLAjGGIGHhVjVmRTBoOGif2-8nY8wN__dBjjEi40sqUGYc4bMphCZE7uP_gddTW3c3g9mBIBSiDTHQ&isca=1) for this pr.

TL;DR:
- Same records, fewer requests (probably because the availability check is gone).
- Catalog is different because of "is_resumable" field now present.
You can see an extended review in previous comments from when I ran the regression locally.

**Comment by aldogonzalez8** [comment]
And the [CAT](https://ff4672dba71969e9a8ec1be85ec41d57bb26a63a94f3bb2a5579bee-apidata.googleusercontent.com/download/storage/v1/b/airbyte-ci-reports-multi/o/airbyte-ci%2Fconnectors%2Ftest%2Fpull_request%2Faldogonzalez8_source_google-analytics-data-api_upgrade-cdk-3%2F1722506761%2F952c34b002c8bfb9abd0ad1574c222adc3f1f5a3%2Fsource-google-analytics-data-api%2F2.4.15%2Foutput.html?jk=AYBlUPCPz26L_axp1Uelx98XxSD2qiJRPAC6vGiDliskEGa7P-L1RPQoGj2rqLoHS1UMQwM9hSf_USUO6zoD4wg5twOaMhczcMD1TKQTAAYLQo9pa0WYyurHN5cK9y90mNqIQXPCDTCKkPSEEkatK5JFCRQVbQIuskybOE4pJ_86Pd2DRSWV9Y6xce8vbIbYk8jbktISX0LHPoWsNSWaV4hErjrhQQ_ExOErt1j036hb0q-zrzBKTc_vwNMqCswLirUK7iwCJoXjkHFdgH5DKaYc8N3loFnFNMVM-3H8LC_VzZUPlaKREqBqYL9SCowiOCGqE7c3kNTM65xbihrvdtP20Ky5SKzN4i-AdeKuZsCEfihW8dIlotarApXIXF-TtIVs4vFCbxeUlVg7hYeFWlXMaoU0TTUZgGwno8_ntWw545vZ4eTJsJ-OlMSJlpWYiOgPQaFVhwGHqzBUSrBEhh7lquxBhHp8mPVTVRvY1qyiuqMlTPhy7PYd3NZ9h-OUuL9RlQjauwkwpQxPHKyAvKq3uOSILd45BsN2qPjaA-rSPxhEmTpspDthTBPhFEoHUXPKZOi0XUDNkmdAmLMmlEgNSlLoCpl41NnRD04MbZDjINsJjJwYJcxLCIIAeMmfBKaQZ9M-wHLR7t8ttmK5jkZ6Q1ZG6nyTYfYKEWRYcbeRanl9DZMJCGmD8uCK-byEofNSm0a8adsgPHYxHkXkWgah68YXqLEv6dcKF3QQtTBGqgtp7bIxn_pvWKiodwruWbUxWhu9x3ucRUMXdfkid9nhaVGnv5lSKeyjrCD2yxUY-RwtfqoyhJyZusEFqDxEHkIo32-FlyS0z_WbSev2hndE5usKmolpX1D2HGNddRPIHFZO_hdioA_xPqhVa8Lz3-1S0B5UrEbGYoqJxRO0VaWsQ9Q2AUwk5U0a6rzTsAvU4at4iVXVHQjocEBCQVE3GLCBnpD9TyCSsxlRM-vpto4lL-4e6Y00gLX672d6aVR_xX6iocR0PBQAAMJK72xtAiMahoOkjTqYAkStbtBnFi93AK3apzbg6-imluXCT6NIH73O5E5nN21C7HR7JfZjm66fnCzE-PoKH6yjoiSRKIEPdnp0idZ0n95VyPgSC1UIUCdly9XMdYSUDFSxZYO9RFVvpWzKdpylk0oJMARBnLy17om_Ia3-yLO1xXl008Hcpl_L34NyxS3JmHSShJ1jDHyWKtS-hhmChVABIdr2TePt4A2k1WV-Edne0Lj0cquNaLnwAVim_tvzHCoJCedgOe1RnYWVvU3o9BQpUtV_ZWZ62fgKS_XwegQdq5GwRDWz_5JQNAgXCra1Tg3-HkUnPxNRMH0WNe0QTxoUOXgfwytZBjevbCcTLFfzndVfqu21GahDVtPYAfZKePytVqOHVVWHBcFskhWsKR-7XFMrVv6Bq807tMdOKs0psCyN4fgAv3WQFefXsKqZoRA&isca=1) is finally passed :)

**Review by bazarnov — APPROVED** [review]
Looks good! Thanks @aldogonzalez8

**Comment by aldogonzalez8** [comment]
I had to update the release information... waiting again for tests.

**Comment by aldogonzalez8** [comment]
This will be released on Aug 5, 2023

**Comment by aldogonzalez8** [comment]
rerunning CI tests

**Comment by aldogonzalez8** [comment]
running again CI tests

**Comment by aldogonzalez8** [comment]
Running ci tests

**Comment by aldogonzalez8** [comment]
running CI tests

**Comment by aldogonzalez8** [comment]
:-/
image
